lib/sierra-models/base.js

```diff
@@ -20,110 +21,95 @@ class SierraBase {
-        // only match (or undefined if for some reason the link is broken):
-        const [parallelValue] = parallelField
-        const directionControlPrefix = direction === 'rtl' ? '\u200F' : ''
-        return {
-          ...parallelValue,
-          value: directionControlPrefix + parallelValue.value,
-          direction,
-          script: scriptCode
+  _parseSubfield6 (field) {
+    // otherwise, it is a raw field from _varField
+    const subfields = field.subfields || field.subFields
+    const subfield6 = subfields?.find((sub) => sub.tag === '6')
+    const content = subfield6?.content
+    if (content) {
+      const parsed = this._parseParallelFieldLink(content)
+      return parsed
+    }
+  }
+
+  // This function takes a marc tag, and returns an array of primary values with
+  //   relevant parallel values attached and or orphaned parallels relevant to
+  //   given marc tag
+  _addParallels (marc) {
+    // _varFieldByMarcTag returns array of raw marc fields filtered by marc tag:
+    //  [{fieldTag: 'a', marcTag: '123', content: 'the thing'}]
+    const primaryRawFields = this._varFieldByMarcTag(marc)
+    const parallelFieldLinks = Object.keys(this.parallelsCache)
+      .filter((parallelLink) => parallelLink.includes(marc))
+    const parallelRawFields = parallelFieldLinks.map((parallelLink) => this.parallelsCache[parallelLink])
+
+    // build lookup table with extracted values
+    const parallelsLookup = parallelRawFields.reduce((lookup, rawField, i) => {
+      const { number } = this._parseParallelFieldLink(parallelFieldLinks[i])
+      // number is the two digits after the dash (100-01, number = 01)
+      //   since we've filtered on marc tag, we can check for match based
+      //   on number, which is the only value available in both primary and
+      //   parallel subfield 6.
+      return { ...lookup, [number]: rawField }
+    }, {})
+
+    const lookup = (field) => {
+      const subfieldSix = this._parseSubfield6(field)
+      if (subfieldSix) {
+        const { number: subfieldSixSuffix } = subfieldSix
+        // if its in table,
+        if (parallelsLookup[subfieldSixSuffix]) {
+          //    add parallel to the primary
+          field.parallel = { ...parallelsLookup[subfieldSixSuffix] }
+          //    set value to null, so we know that parallel has been matched
+          parallelsLookup[subfieldSixSuffix] = null
         }
-      })
-    return parallelValues
+      }
+    }
+
+    const primariesWithAndWithoutMatches = primaryRawFields.map((field, i) => {
+      // attach relevant parallels
+      lookup(field)
+      return field
+    })
+
+    const orphanParallels = parallelRawFields.map((field) => {
+      // check if parallel remains unmatched in lookup table.
+      //   if it is, it will be returned {parallel: {value:'meep', subfieldMap: {'a': 'meep'}}}
+      lookup(field)
+      if (field.parallel) {
+        return field
+      } else { return null }
```

Comment:
Something in here isn't right. An orphaned parallel is rendered as
```
{
  value: "parallel value",
  ...
  parallel: {
    value: "parallel value",
    ...
  }
}
```
See [this pr](https://github.com/NYPL/research-catalog-indexer/pull/20) with some suggestions to address this bug and clarify a few things